includes/gateways/stripe/includes/class-give-stripe-customer.php

```diff
@@ -254,18 +254,39 @@ public function update_customer( $id, $args ) {
 	 * @param string $customer_id Customer ID provided by Stripe.
 	 *
 	 * @since 2.5.10
+	 * @see https://stripe.com/docs/api/payment_methods/attach
 	 *
 	 * @return \Stripe\Customer
 	 */
 	public function set_default_payment_method( $id, $customer_id ) {
+		$customer = null;
 
-		$update_args = array(
-			'invoice_settings' => array(
-				'default_payment_method' => $id,
-			),
-		);
+		try{
+			$payment_method = $this->stripe_gateway->payment_method->retrieve( $id );
+
+			// Add card only if not added before.
+			if( $customer_id !== $payment_method->customer ){
+				$payment_method->attach(array(
+					'customer' => $customer_id
+				));
+			}
+
+			$update_args = array(
+				'invoice_settings' => array(
+					'default_payment_method' => $id,
+				),
+			);
 
-		return $this->update_customer( $customer_id, $update_args );
+			$customer = $this->update_customer( $customer_id, $update_args );
+
+		}catch( Exception $e ){
+
+			give_set_error( 'stripe_error', $e->getMessage() );
+			give_record_gateway_error( __( 'Stripe Payment Method Error', 'give' ), sprintf( "%s\n%s", __( 'An error occurred retrieving or creating the ', 'give' ), $e->getMessage() ) );
```

Comment:
@ravinderk I don't think you finished writing the proper error explanation here. I noticed this after I merged. Please fix and commit to `develop`.